### ISSUE TITLE
upgrade from ansible-core 2.11.2 to 2.11.3+ causes failure:  (plugin_type: connection plugin: eci setting: reconnection_retries ) was not defined in configuration.

### ISSUE THREAD

**Issue by charltonstanley** [mpieters3/ansible-eci-connector#19]
Here is a successful connection attempt before `ansible-core` was updated:
```
PS /home/linux_user/svn/ansible/playbook> ansible all -i ./env/aws/inventory.yml -m command -a /usr/bin/hostname -c eci -vvv
ansible [core 2.11.2] 
  config file = /home/linux_user/svn/ansible/playbook/ansible.cfg
  configured module search path = ['/home/linux_user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/linux_user/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/linux_user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/linux_user/.local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 2.10.1
  libyaml = True
Using /home/linux_user/svn/ansible/playbook/ansible.cfg as config file
host_list declined parsing /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml as it did not pass its verify_file() method
script declined parsing /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml as it did not pass its verify_file() method
Parsed /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml inventory source with yaml plugin
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
META: ran handlers
TEMPORARY KEY LOCATION: /tmp/tmpaorxn2x7
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
ECI PUB KEY EXPIRING/NOT SENT, PUSHING NOW
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b 10.1.2.251 '/bin/sh -c '"'"'echo ~ec2-user && sleep 0'"'"''
<10.1.2.251> (0, b'/home/ec2-user\n', b'')
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b 10.1.2.251 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/ec2-user/.ansible/tmp `"&& mkdir "` echo /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002 `" && echo ansible-tmp-1644512793.8184526-316390-99620099136002="` echo /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002 `" ) && sleep 0'"'"''
<10.1.2.251> (0, b'ansible-tmp-1644512793.8184526-316390-99620099136002=/home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002\n', b'')
<ec2-hostname> Attempting python interpreter discovery
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b 10.1.2.251 '/bin/sh -c '"'"'echo PLATFORM; uname; echo FOUND; command -v '"'"'"'"'"'"'"'"'/usr/bin/python'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.9'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.8'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.7'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.6'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python3.5'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python2.7'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python2.6'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'/usr/libexec/platform-python'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'/usr/bin/python3'"'"'"'"'"'"'"'"'; command -v '"'"'"'"'"'"'"'"'python'"'"'"'"'"'"'"'"'; echo ENDFOUND && sleep 0'"'"''
<10.1.2.251> (0, b'PLATFORM\nLinux\nFOUND\n/usr/bin/python\n/usr/bin/python3.7\n/usr/bin/python2.7\n/usr/bin/python3\n/usr/bin/python\nENDFOUND\n', b'')
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b 10.1.2.251 '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<10.1.2.251> (0, b'{"osrelease_content": "NAME=\\"Amazon Linux\\"\\nVERSION=\\"2\\"\\nID=\\"amzn\\"\\nID_LIKE=\\"centos rhel fedora\\"\\nVERSION_ID=\\"2\\"\\nPRETTY_NAME=\\"Amazon Linux 2\\"\\nANSI_COLOR=\\"0;33\\"\\nCPE_NAME=\\"cpe:2.3:o:amazon:amazon_linux:2\\"\\nHOME_URL=\\"https://amazonlinux.com/\\"\\n", "platform_dist_result": ["", "", ""]}\n', b'')
<ec2-hostname> Python interpreter discovery fallback (unsupported Linux distribution: amzn)
Using module file /home/linux_user/.local/lib/python3.8/site-packages/ansible/modules/command.py
<10.1.2.251> PUT /home/linux_user/.ansible/tmp/ansible-local-316386b6wp0v3o/tmprv2apv3c TO /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002/AnsiballZ_command.py
<10.1.2.251> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b '[10.1.2.251]'
<10.1.2.251> (0, b'sftp> put /home/linux_user/.ansible/tmp/ansible-local-316386b6wp0v3o/tmprv2apv3c /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002/AnsiballZ_command.py\n', b'')
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b 10.1.2.251 '/bin/sh -c '"'"'chmod u+x /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002/ /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002/AnsiballZ_command.py && sleep 0'"'"''
<10.1.2.251> (0, b'', b'')
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b -tt 10.1.2.251 '/bin/sh -c '"'"'/usr/bin/python /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002/AnsiballZ_command.py && sleep 0'"'"''
<10.1.2.251> (0, b'\r\n{"changed": true, "end": "2022-02-10 11:06:35.333641", "stdout": "ec2-hostname.ec2.internal", "cmd": ["/usr/bin/hostname"], "rc": 0, "start": "2022-02-10 11:06:35.327059", "stderr": "", "delta": "0:00:00.006582", "invocation": {"module_args": {"creates": null, "executable": null, "_uses_shell": false, "strip_empty_ends": true, "_raw_params": "/usr/bin/hostname", "removes": null, "argv": null, "warn": false, "chdir": null, "stdin_add_newline": true, "stdin": null}}}\r\n', b'Shared connection to 10.1.2.251 closed.\r\n')
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
<10.1.2.251> SSH: EXEC ssh -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=22 -o 'IdentityFile="/tmp/tmpaorxn2x7"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o 'User="ec2-user"' -o ConnectTimeout=10 -o ControlPath=/home/linux_user/.ansible/cp/d487c7ea5b 10.1.2.251 '/bin/sh -c '"'"'rm -f -r /home/ec2-user/.ansible/tmp/ansible-tmp-1644512793.8184526-316390-99620099136002/ > /dev/null 2>&1 && sleep 0'"'"''
<10.1.2.251> (0, b'', b'')
[WARNING]: Platform linux on host ec2-hostname is using the discovered Python interpreter at /usr/bin/python, but future installation of another Python interpreter could change the meaning of that path.
See https://docs.ansible.com/ansible/2.11/reference_appendices/interpreter_discovery.html for more information.
ec2-hostname | CHANGED | rc=0 >>
ec2-hostname.ec2.internal
META: ran handlers
META: ran handlers
```

After `ansible-core` was update to `2.11.3`, I received `ansible.errors.AnsibleError: Requested entry (plugin_type: connection plugin: eci setting: reconnection_retries ) was not defined in configuration.`. Detailed output is below.

```
PS /home/linux_user/svn/ansible/playbook> ansible all -i ./env/aws/inventory.yml -m command -a /usr/bin/hostname -c eci -vvvv  
ansible [core 2.11.3] 
  config file = /home/linux_user/svn/ansible/playbook/ansible.cfg
  configured module search path = ['/home/linux_user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/linux_user/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/linux_user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/linux_user/.local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 2.10.1
  libyaml = True
Using /home/linux_user/svn/ansible/playbook/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml as it did not pass its verify_file() method
script declined parsing /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml as it did not pass its verify_file() method
Parsed /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml inventory source with yaml plugin
Loading callback plugin minimal of type stdout, v2.0 from /home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/callback/minimal.py
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
META: ran handlers
TEMPORARY KEY LOCATION: /tmp/tmp7hrxsj_g
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
The full traceback is:
Traceback (most recent call last):
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/__init__.py", line 58, in get_option
    option_value = C.config.get_config_value(option, plugin_type=get_plugin_class(self), plugin_name=self._load_name, variables=hostvars)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/config/manager.py", line 438, in get_config_value
    value, _drop = self.get_config_value_and_origin(config, cfile=cfile, plugin_type=plugin_type, plugin_name=plugin_name,
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/config/manager.py", line 571, in get_config_value_and_origin
    raise AnsibleError('Requested entry (%s) was not defined in configuration.' % to_native(_get_entry(plugin_type, plugin_name, config)))
ansible.errors.AnsibleError: Requested entry (plugin_type: connection plugin: eci setting: reconnection_retries ) was not defined in configuration.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/executor/task_executor.py", line 158, in run
    res = self._execute()
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/executor/task_executor.py", line 582, in _execute
    result = self._handler.run(task_vars=variables)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/command.py", line 26, in run
    results = merge_hash(results, self._execute_module(module_name='ansible.legacy.command', task_vars=task_vars, wrap_async=wrap_async))
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 969, in _execute_module
    self._make_tmp_path()
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 389, in _make_tmp_path
    tmpdir = self._remote_expand_user(self.get_shell_option('remote_tmp', default='~/.ansible/tmp'), sudoable=False)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 852, in _remote_expand_user
    data = self._low_level_execute_command(cmd, sudoable=False)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 1264, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/usr/share/ansible/plugins/connection/eci.py", line 402, in exec_command
    return ssh.Connection.exec_command(self, cmd=cmd, in_data=in_data, sudoable=sudoable)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/connection/ssh.py", line 1268, in exec_command
    (returncode, stdout, stderr) = self._run(cmd, in_data, sudoable=sudoable)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/connection/ssh.py", line 441, in wrapped
    remaining_tries = int(self.get_option('reconnection_retries')) + 1
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/__init__.py", line 60, in get_option
    raise KeyError(to_native(e))
KeyError: 'Requested entry (plugin_type: connection plugin: eci setting: reconnection_retries ) was not defined in configuration.'
ec2-hostname | FAILED | rc=-1 >>
Unexpected failure during module execution.
PS /home/linux_user/svn/ansible/playbook> 
```

Out of curiosity, I decided to try updating `ansible-core` the current latest version of `2.12.2`. I received a similar but different error this time: `ansible.errors.AnsibleError: Requested entry (plugin_type: connection plugin: eci setting: pkcs11_provider ) was not defined in configuration.`. Detailed output is below.

```
PS /home/linux_user/svn/ansible/playbook> ansible all -i ./env/aws/inventory.yml -m command -a /usr/bin/hostname -c eci -vvv
ansible [core 2.12.2]
  config file = /home/linux_user/svn/ansible/playbook/ansible.cfg
  configured module search path = ['/home/linux_user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/linux_user/.local/lib/python3.8/site-packages/ansible
  ansible collection location = /home/linux_user/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/linux_user/.local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
  jinja version = 2.10.1
  libyaml = True
Using /home/linux_user/svn/ansible/playbook/ansible.cfg as config file
host_list declined parsing /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml as it did not pass its verify_file() method
script declined parsing /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml as it did not pass its verify_file() method
Parsed /home/linux_user/svn/ansible/playbook/env/aws/inventory.yml inventory source with yaml plugin
Skipping callback 'default', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
META: ran handlers
TEMPORARY KEY LOCATION: /tmp/tmplnsbd2lo
<10.1.2.251> ESTABLISH SSH CONNECTION FOR USER: ec2-user
The full traceback is:
Traceback (most recent call last):
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/__init__.py", line 58, in get_option
    option_value = C.config.get_config_value(option, plugin_type=get_plugin_class(self), plugin_name=self._load_name, variables=hostvars)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/config/manager.py", line 425, in get_config_value
    value, _drop = self.get_config_value_and_origin(config, cfile=cfile, plugin_type=plugin_type, plugin_name=plugin_name,
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/config/manager.py", line 558, in get_config_value_and_origin
    raise AnsibleError('Requested entry (%s) was not defined in configuration.' % to_native(_get_entry(plugin_type, plugin_name, config)))
ansible.errors.AnsibleError: Requested entry (plugin_type: connection plugin: eci setting: pkcs11_provider ) was not defined in configuration.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/executor/task_executor.py", line 158, in run
    res = self._execute()
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/executor/task_executor.py", line 589, in _execute
    result = self._handler.run(task_vars=variables)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/command.py", line 26, in run
    results = merge_hash(results, self._execute_module(module_name='ansible.legacy.command', task_vars=task_vars, wrap_async=wrap_async))
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 973, in _execute_module
    self._make_tmp_path()
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 390, in _make_tmp_path
    tmpdir = self._remote_expand_user(self.get_shell_option('remote_tmp', default='~/.ansible/tmp'), sudoable=False)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 856, in _remote_expand_user
    data = self._low_level_execute_command(cmd, sudoable=False)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/action/__init__.py", line 1253, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/usr/share/ansible/plugins/connection/eci.py", line 402, in exec_command
    return ssh.Connection.exec_command(self, cmd=cmd, in_data=in_data, sudoable=sudoable)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/connection/ssh.py", line 1306, in exec_command
    cmd = self._build_command(ssh_executable, 'ssh', *args)
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/connection/ssh.py", line 647, in _build_command
    pkcs11_provider = self.get_option("pkcs11_provider")
  File "/home/linux_user/.local/lib/python3.8/site-packages/ansible/plugins/__init__.py", line 60, in get_option
    raise KeyError(to_native(e))
KeyError: 'Requested entry (plugin_type: connection plugin: eci setting: pkcs11_provider ) was not defined in configuration.'
ec2-hostname | FAILED | rc=-1 >>
Unexpected failure during module execution.
PS /home/linux_user/svn/ansible/playbook> 
```

I'm not quite sure what to make of this. I tried debugging it a little but I didn't really get anywhere, as I am pretty green when it comes to python.  Please let me know if you need any additional info; happy to provide it.

**Comment by charltonstanley** [comment]
Looking into it further, I believe it is due to some bugfixes that ansible made, that changed some of the connection parameters that this eci plugin relies on.  Below is what I believe this is what caused the failure in `2.11.3`.

<https://github.com/ansible/ansible/blob/v2.11.3/changelogs/CHANGELOG-v2.11.rst>

> ### Bugfixes
>...
>* ssh_connection - rename retries to reconnection_retries to avoid conflicts with task vars (https://github.com/ansible/ansible/issues/75142).
>* ssh_connection - set the default for reconnection_retries back to 0 (https://github.com/ansible/ansible/issues/75142).
>* task_executor/ssh_connection - use the retries value from ssh_connection settings, not the default from the Task field attributes (https://github.com/ansible/ansible/issues/75142).

Below is what I believe this is what caused the failure in `2.12.2`.

<https://github.com/ansible/ansible/blob/v2.12.0/changelogs/CHANGELOG-v2.12.rst>

> ### Minor changes
>...
>* ssh - added pkcs11 support by adding the pkcs11_provider option in the ssh connection module. (https://www.github.com/ansible/ansible/pull/32829)

**Comment by calscoo** [comment]
It looks like renaming the `retries`  option to `reconnection_retries` and leaving the keys alone in the options documentation  like so
```
      reconnection_retries:
          # constant: ANSIBLE_SSH_RETRIES
          description: Number of attempts to connect.
          default: 3
          type: integer
          env:
            - name: ANSIBLE_SSH_RETRIES
          ini:
            - section: connection
              key: retries
            - section: ssh_connection
              key: retries
          vars:
            - name: ansible_ssh_retries
              version_added: '2.7'
```
fixes the bug in 2.11.3. Adding an additional `pkcs11_provider` option documentation block like so
```
      pkcs11_provider:
          version_added: '2.12'
          default: ""
          description:
            - "PKCS11 SmartCard provider such as opensc, example: /usr/local/lib/opensc-pkcs11.so"
            - "Requires sshpass version 1.06+, sshpass must support the -P option."
          env: [{name: ANSIBLE_PKCS11_PROVIDER}]
          ini:
            - {key: pkcs11_provider, section: ssh_connection}
          vars:
            - name: ansible_ssh_pkcs11_provider
```
resolves the bug in 2.12.2. However, I'm unaware if adding the option for `pkcs11_provider` without supporting it in the script is going to cause problems. That being said, a large number of the options listed in the documentation for this script are not referenced within the script.

**Comment by mpieters3** [comment]
Resolved in https://github.com/mpieters3/ansible-eci-connector/releases/tag/v2.12.0
